matter_server/server/device_controller.py

```diff
@@ -1164,52 +1194,30 @@ async def log_node_long_setup(time_start: float) -> None:
-                    except NodeInterviewFailed as err:
-                        node_logger.warning(
-                            "Setup for node failed: %s",
-                            str(err) or err.__class__.__name__,
-                            # log full stack trace if verbose logging is enabled
-                            exc_info=err
-                            if LOGGER.isEnabledFor(VERBOSE_LOG_LEVEL)
-                            else None,
-                        )
-                        # NOTE: the node will be picked up by mdns discovery automatically
-                        # when it comes available again.
-                        return
-
-                # setup subscriptions for the node
-                try:
-                    await self._subscribe_node(node_id)
-                except ChipStackError as err:
-                    node_logger.warning(
-                        "Unable to subscribe to Node: %s",
-                        str(err) or err.__class__.__name__,
-                        # log full stack trace if verbose logging is enabled
-                        exc_info=err
-                        if LOGGER.isEnabledFor(VERBOSE_LOG_LEVEL)
-                        else None,
-                    )
-                    # NOTE: the node will be picked up by mdns discovery automatically
-                    # when it becomes available again.
-                    return
+            # setup subscriptions for the node
+            try:
+                await self._subscribe_node(node_id)
+            except ChipStackError as err:
+                node_logger.warning(
+                    "Unable to subscribe to Node: %s",
+                    str(err) or err.__class__.__name__,
+                    # log full stack trace if verbose logging is enabled
+                    exc_info=err if LOGGER.isEnabledFor(VERBOSE_LOG_LEVEL) else None,
+                )
+                # NOTE: the node will be picked up by mdns discovery automatically
+                # when it becomes available again.
+                return
 
-                # check if this node has any custom clusters that need to be polled
-                if polled_attributes := check_polled_attributes(node_data):
-                    self._polled_attributes[node_id] = polled_attributes
-                    self._schedule_custom_attributes_poller()
+            # check if this node has any custom clusters that need to be polled
+            if polled_attributes := check_polled_attributes(node_data):
+                self._polled_attributes[node_id] = polled_attributes
+                self._schedule_custom_attributes_poller()
 
-            finally:
-                log_timers[node_id].cancel()
-                self._nodes_in_setup.discard(node_id)
+        finally:
+            log_timers[node_id].cancel()
+            self._nodes_in_setup.discard(node_id)
+            if is_thread_node:
+                self._thread_node_setup_throttle.release()
```

Comment:
We potentially double release a semaphore here when the node was stuck previously but then gets unstuck no?

Reply:
Yeah, correct but that doesn't seem to harm as its just a counter